src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
 
     Returns
     -------
-    dict
-        The BOM content as a dictionary.
+    Bom
+        The CycloneDX Bom object.
 
     Raises
     ------
     CycloneDXParserError
         If the bom.json file cannot be located or deserialized.
-
     """
-    # TODO: use the official CycloneDX library to validate and deserialize BOM files
-    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
     if not os.path.exists(file_path):
         raise CycloneDXParserError(f"Unable to locate any BOM files at: {str(file_path.parent)}.")
 
+    # We use the `cyclonedx-python-library` library for deserialization following the example here:
+    # https://cyclonedx-python-library.readthedocs.io/en/v7.3.4/examples.html
     with open(file_path, encoding="utf8") as file:
+        json_data = file.read()
+        if defaults.getboolean("dependency.resolver", "validate", fallback=True):
+            try:
+                schema_version = defaults.get("dependency.resolver", "schema", fallback="1.6")
```

Comment:
I wonder if we are intending to catch `ValueError` from `defaults.get(...)` too?. If not, I think this line can be moved outside of the try catch block.